pubsub/awssnssqs/awssnssqs.go

```diff
@@ -186,9 +186,9 @@ func (o *URLOpener) OpenTopicURL(ctx context.Context, u *url.URL) (*pubsub.Topic
 	if err != nil {
 		return nil, fmt.Errorf("open topic %v: %v", u, err)
 	}
-	client := sns.New(o.ConfigProvider, overrideCfg)
+	o.TopicOptions.ClientConfig = overrideCfg
```

Comment:
See https://github.com/google/go-cloud/blob/master/blob/s3blob/s3blob.go#L139 for an example of how to overlay `overrideCfg` on top of `o.ConfigProvider` without using `Options`.